esp/esp/program/modules/handlers/teacherbigboardmodule.py

```diff
@@ -0,0 +1,163 @@
+        }
+        return render_to_response('program/modules/bigboardmodule/bigboard.html',
+                                  request, context)
+
+    # Numbers computed for the big board are below.  They're cached for 105
+    # seconds, which is long enough that they hopefully won't get recomputed a
+    # bunch if multiple admins are loading the page, but short enough that each
+    # time the page refreshes for the same admin, they will get new numbers
+
+    @cache_function_for(105)
+    def num_teachers_teaching(self, prog):
+        # Querying for SRs and then extracting the users saves us joining the
+        # users table.
+        return ClassSubject.objects.filter(parent_program=prog
+        ).values_list('teachers').distinct().count()
+
+    @cache_function_for(105)
+    def num_active_users(self, prog, minutes=10):
+        recent = datetime.datetime.now() - datetime.timedelta(0, minutes * 60)
+        return ClassSubject.objects.filter(parent_program=prog, timestamp__gt=recent
+        ).values_list('teachers').distinct().count()
+
+    @cache_function_for(105)
+    def num_checked_in_teachers(self, prog):
+        return Record.objects.filter(program=prog, event='teacher_checked_in').count()
+
+    @cache_function_for(105)
+    def num_class_reg(self, prog):
+        return ClassSubject.objects.filter(parent_program=prog).values_list('id').count()
+
+    @cache_function_for(105)
+    def num_class_app(self, prog):
+        return ClassSubject.objects.filter(parent_program=prog, status__gt=0, sections__status__gt=0).count()
+
+    @cache_function_for(105)
+    def reg_classes(self, prog):
+        class_times = ClassSubject.objects.filter(parent_program=prog
+        ).values_list('timestamp', flat=True)
+        return sorted(class_times)
+
+    @cache_function_for(105)
+    def app_classes(self, prog):
+        #all ClassSubjects that are approved (and have an approved section)
+        class_times = ClassSubject.objects.filter(parent_program=prog, status__gt=0, sections__status__gt=0
+        ).values_list('timestamp', flat=True)
+        return sorted(class_times)
+
+    @cache_function_for(105)
+    def teach_times(self, prog):
+        teacher_times = dict(ClassSubject.objects.filter(parent_program=prog
+        ).values_list('teachers').annotate(Min('timestamp')))
+        return sorted(teacher_times.itervalues())
+
+    @cache_function_for(105)
+    def get_hours(self, prog):
+        hours = ClassSubject.objects.filter(parent_program=prog
+        ).exclude(category__category__iexact="Lunch").values_list('timestamp','duration','class_size_max')
+        sorted_hours = sorted(hours, key=operator.itemgetter(2))
+        class_hours = [(hour[1],hour[0]) for hour in sorted_hours]
+        student_hours = [(hour[1]*hour[2], hour[0]) for hour in sorted_hours]
+        return class_hours, student_hours
+
+    @cache_function_for(105)
+    def static_hours(self, prog):
```

Comment:
This does the same thing as `get_hours` that just extracts only some of the data, right?  Can it just be a wrapper around that that extracts the relevant data?